From 130ce698f95e7295deaf51791ddd1910afcc8bf3 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Fri, 8 May 2020 22:33:55 -0700
Subject: [PATCH 0498/1921] octeontx2-af: Fix reading SSOW_LF_GWS_TAG after
 rvu_poll_reg()

The commit "octeontx2-af: Make SSO/SSOW LF teardown less CPU intensive"
introduced a bug. The value read from SSOW_LF_GWS_TAG is used in next
statement. So this patch fixes by reading SSOW_LF_GWS_TAG again.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: Iefd0768ce81f6fdd807f0d67dd882ba2628b17db
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28144
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
index 0ff69461de8c..d3617acfa72a 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
@@ -299,6 +299,8 @@ int rvu_sso_lf_teardown(struct rvu *rvu, u16 pcifunc, int lf, int slot)
 	rvu_poll_reg(rvu, ssow_blkaddr, SSOW_AF_BAR2_ALIASX(0, SSOW_LF_GWS_TAG),
 		     BIT_ULL(63), true);
 
+	reg = rvu_read64(rvu, ssow_blkaddr,
+			 SSOW_AF_BAR2_ALIASX(0, SSOW_LF_GWS_TAG));
 	if (reg & BIT_ULL(62))
 		rvu_write64(rvu, ssow_blkaddr,
 			    SSOW_AF_BAR2_ALIASX(0, SSOW_LF_GWS_OP_DESCHED), 0);
-- 
2.31.1

